proto/storage/cluster.proto

```diff
@@ -94,6 +95,56 @@ message AdmissionControllerConfig {
   bool enforce_on_updates = 5;
 }
 
+message ProcessConfig {
+  bool enabled = 1;
+}
+
+message NetworkConnectionConfig {
+  bool enabled = 1;
+  bool aggregate_external = 2;
+}
+
+message NetworkEndpointConfig {
+  bool enabled = 1;
+  bool include_listening_endpoint_processes = 2;
+}
+
+message CollectorFeature {
+  oneof feature {
+    ProcessConfig processes = 1;
+    NetworkConnectionConfig network_connections = 2;
+    NetworkEndpointConfig network_endpoints = 3;
+  }
+}
+
+message CollectorNamespaceConfig {
+  message NamespaceRule {
+    string namespace = 1;
+    storage.MatchType match_type = 2;
+  }
+
+  CollectorFeature feature = 1;
+  repeated NamespaceRule namespace_selection = 2;
+}
+
+// CollectorConfig controls which type of data is reported by collector
+// and processing of that data by collector. For now this configuration
+// controls if processes, network connections, and network endpoints are
+// reported. It also controls if process information is reported with
+// network endpoints and if external IPs are aggregated. External IP
+// aggregation is controlled at the namespace level and in the future
+// other features may also be controlled at the namespace level.
+message CollectorConfig {
+  // The following three are applied at the cluster level.
+  ProcessConfig process_config = 1;
+  NetworkConnectionConfig network_connection_config = 2;
+  NetworkEndpointConfig network_endpoint_config = 3;
+  // The namespace scope config is an array since it controls multiple
+  // features and for each feature there can be different configs for
+  // different sets of namespaces.
```

Comment:
I think this is too concrete, when a new option is added one must adjust this doc to `four`. Instead it could be a paragraph like:

```suggestion
  // The following configs are applied at the cluster level.
  ProcessConfig process_config = 1;
  NetworkConnectionConfig network_connection_config = 2;
  NetworkEndpointConfig network_endpoint_config = 3;
  
  // The namespace scope config is an array since it controls multiple
  // features and for each feature there can be different configs for
  // different sets of namespaces.
```

Reply:
Done